File: Data Structures & Algorithms/longest-turbulent-subarray/submission-3.py
class Solution:
    def maxTurbulenceSize(self, arr: List[int]) -> int:
        prev=""
        l=0
        r=1
        n=len(arr)
        res=1
        while r<n:
            if arr[r]>arr[r-1] and prev!=">":
                res=max(res,r-l+1)
                r+=1
                prev=">"
                print(l,r)
            elif arr[r]<arr[r-1] and prev!="<":
                res=max(res,r-l+1)
                r+=1
                prev="<"
                print(l,r)
            else:
                if arr[r]==arr[r-1]:
                    r+=1
                l=r-1
                prev=""
            
        return res

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int l =0;
        int r =0;
        Set<Character> set = new HashSet<>();
        int max = 0;
        char[] arr = s.toCharArray();
        while(r < arr.length){
            if(!set.contains(arr[r])){
                set.add(arr[r]);
                r++;
                max = Math.max(max, r-l);
            }else{
                set.remove(arr[l]);
                l++;
            }
        }
        return max;
    }
}
